src/commands/path/path.generate.ts

```diff
@@ -0,0 +1,214 @@
+    const region = getRegion(collection);
+    const event = getEvent(collection);
+    const date = getDate(collection);
+    const gsd = extractGsd(tiff);
+    const epsg = extractEpsg(tiff);
+
+    const target = generatePath(args.targetBucketName, category, geographicDescription, region, event, date, gsd, epsg);
+    logger.info({ duration: performance.now() - startTime, target: target }, 'GeneratePath:Done');
+    return target;
+  },
+});
+
+/**
+ * Generates Target Path based on category
+ *
+ * @param {string} targetBucketName
+ * @param {string} category
+ * @param {string} geospatial_description
+ * @param {string} region
+ * @param {string} event
+ * @param {string} date
+ * @param {string} gsd
+ * @param {string} epsg
+ * @returns {string}
+ */
+export function generatePath(
+  targetBucketName: string,
+  category: string,
+  geospatial_description: string,
+  region: string,
+  event: string,
+  date: string,
+  gsd: string,
+  epsg: string,
+): string {
+  const name = generateName(region, geospatial_description, event);
+
+  if (category === dataCategories.SCANNED_AERIAL_PHOTOS) {
+    // nb: Historic Imagery is out of scope as survey number is not yet recorded in collection metadata
+    throw new Error(`Automated target generation not implemented for historic imagery`);
+  } else if ([dataCategories.URBAN_AERIAL_PHOTOS, dataCategories.RURAL_AERIAL_PHOTOS].includes(category)) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}_${gsd}/rgb/${epsg}/`;
+  } else if (category === dataCategories.SATELLITE_IMAGERY) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}_${gsd}/rgb/${epsg}/`;
+  } else if ([dataCategories.DEM, dataCategories.DSM].includes(category)) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}/${category}_${gsd}/${epsg}/`;
+  } else {
+    throw new Error(`Path Can't be generated from collection as no matching category: ${category}.`);
+  }
+}
+
+/**
+ * Generates specific dataset name based on metadata inputs
+ *
+ * @param {string} region
+ * @param {string} geospatial_description
+ * @param {string} event
+ * @returns {string}
+ */
+export function generateName(region: string, geospatial_description: string, event: string): string {
+  let name = region;
+  if (geospatial_description) {
+    name = geospatial_description.toLowerCase().replace(/\s+/g, '-');
+  }
+  if (event) {
+    name = `${name}-${event.toLowerCase().replace(/\s+/g, '-')}`;
+  }
+  return name;
+}
+
+export function getCategory(collection: StacCollection): string {
+  const category = collection['linz:geospatial_category'] as string;
+  if (!category) {
+    throw new Error('No category in collection');
+  }
+  return category;
+}
+
+export function getGeographicDescription(collection: StacCollection): string {
+  // This is optional metadata, therefore returning nothing is ok.
+  return collection['linz:geographic_description'] as string;
```

Comment:
this is bypassing the typechcker this should return `string | undefined` as the collection may or may not have the value